migrate_pim/upgrade_from_32_to_40.rst

```diff
@@ -0,0 +1,217 @@
+
+
+Updated System components
+-------------------------
+
+You have to make sure your system components are upgrade to the version required for Akeneo PIM 4.0:
+ - PHP 7.3
+ - MySQL 8
+ - Elasticsearch 7.5
+
+
+.. note::
+    MySQL and Elasticsearch support in-place update: MySQL 8 will be able to use database files
+    created by a MySQL 5.7 version and Elasticsearch 7.5 will be able to use indexes created
+    by Elasticsearch 6.
+
+    So there's no need to export and reimport data for these systems.
+
+Upgraded Virtual Host configuration
+-----------------------------------
+
+As Akeneo PIM 4.0 is based on Symfony 4, the Web entry point has been changed:
+ - 3.2: `web/app.php`
+ - 4.0: `public/index.php`
+
+You can check the VirtualHost configuration for 4.0: system_requirements/system_configuration.rst.inc.
+
+
+Prepare your project
+********************
+
+In the following code, "my_pim" is the directory where the Akeneo PIM 3.2 you want to migrate is installed.
+
+.. code:: bash
+
+    $ cd my_pim
+    $ rm -rf var/cache/
+    $ tar --strip-component 1 -xvzf /path/to/pim-enterprise-standard-v4.0.0.tar.gz  pim-enterprise-standard/composer.json
+
+Load your PIM Enterprise 4.0 dependencies
+*****************************************
+
+.. code:: bash
+
+    $ composer update
+
+
+.. note::
+
+    You may need to increase the memory provided to `composer`, as this step can be very memory consuming:
+
+    .. code:: bash
+
+        $ php -d "memory_limit=4G" /path/to/composer
+
+Let Akeneo PIM 4.0 continue the preparation for you
+***************************************************
+
+.. code:: bash
+
+    $ vendor/akeneo/pim-enterprise-dev/std-build/migration/prepare_32_to_40.sh
+
+
+.. note::
+
+    This script will update the current filesystem layout to match Symfony 4 structure.
+    See http://fabien.potencier.org/symfony4-directory-structure.html for more details.
+
+
+MySQL and ES Credentials Access
+*******************************
+
+Akeneo PaaS
+-----------
+
+If you are using an Akeneo PaaS env, the credentials for MySQL and ES are already available as environment variables.
+These environment variables will be directly available to your Akeneo PIM.
+
+Local or on-premise environment
+-------------------------------
+
+You can make the variables content available to your program or set them in a .env.local file:
+
+.. code::
+
+    APP_DATABASE_HOST=mysql-host
+    APP_DATABASE_PORT=3306
+    APP_DATABASE_NAME=akeneo_pim_db_name
+    APP_DATABASE_USER=akeneo_pim_user
+    APP_DATABASE_PASSWORD=akeneo_pim_password
+    APP_INDEX_HOSTS=elasticsearch-host:9200
+
+
+Make sure your environment is ready to be migrated
+**************************************************
+
+.. code:: bash
+
+    $ bin/console pim:installer:check-requirements
+
+
+If this command detects something not working or not properly configured,
+please fix the problem before continuing.
+
+Prepare the front
+*****************
+
+.. code:: bash
```

Comment:
shouldn't we use the `make` commands?

Reply:
I didn't plan any Makefile for the migration. But I can add it.